sql/distinct.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Tamir Duberstein (tamird@gmail.com)
+
+package sql
+
+import (
+	"bytes"
+	"strings"
+
+	"github.com/cockroachdb/cockroach/sql/parser"
+	"github.com/cockroachdb/cockroach/util"
+)
+
+// distinct constructs a distinctNode.
+func (*planner) distinct(n *parser.Select) *distinctNode {
+	if n.Distinct {
+		d := distinctNode{}
+		d.prefixSeen = []byte{}
```

Comment:
No need to allocate an empty `[]byte` here. Just remove this line. The default value of `nil` is fine.